### ISSUE TITLE
chips: md-chip-append-delay treats 0 as the default value

### ISSUE THREAD

**Issue by mgol** [angular/material#10408]
**Actual Behavior**:
 -  `What is the issue? *` md-chip-append-delay="0" has the same result as md-chip-append-delay="300"
 -  `What is the expected behavior?` The append should be immediate.

**CodePen** (or steps to reproduce the issue): *
 -  `CodePen Demo which shows your issue:` https://codepen.io/mgol/pen/BpbJXL
 -  `Details:` Type `b`, accept Broccoli and see the delay happening.

**Angular Versions**: *
 -  `Angular Version:` 1.5.5 (but any version, really)
 -  `Angular Material Version:` 1.1.3

**Additional Information**:
 -  `Browser Type: *` Chrome (others as well, I think)
 -  `Browser Version: *` 56
 -  `OS: *` macOS
 -  `Stack Traces:`

----
Shortcut to create a [new CodePen Demo](http://codepen.io/team/AngularMaterial/pen/bEGJdd).
Note: `*` indicates required information. Without this information, your issue may be auto-closed.

> Do not modify the titles or questions. Simply add your responses to the ends of the questions.
  Add more lines if needed.

The issue is caused by:
https://github.com/angular/material/blob/f18cb2b/src/components/chips/js/chipsController.js#L183
`parseInt(newValue)` for `newValue === 0` is falsy so the default gets used.

The line should be changed to:
```js
var numberValue = parseInt(newValue);
ctrl.chipAppendDelay = isNaN(numberValue) ? DEFAULT_CHIP_APPEND_DELAY : numberValue;
```

**Comment by matt-snider** [comment]
As a temporary workaround we are using a small non-zero number for the delay (e.g. 5)
